arbnode/transaction_streamer.go

```diff
@@ -968,7 +968,15 @@ func (s *TransactionStreamer) executeNextMsg(ctx context.Context, exec execution
 		log.Error("feedOneMsg failed to readMessage", "err", err, "pos", pos)
 		return false
 	}
-	err = s.exec.DigestMessage(pos, msg)
+	var msgForPrefetch *arbostypes.MessageWithMetadata
+	if pos+1 < msgCount {
+		msgForPrefetch, err = s.GetMessage(pos + 1)
+		if err != nil {
+			log.Error("feedOneMsg failed to readMessage", "err", err, "pos", pos+1)
+			return false
+		}
```

Comment:
Error from here is leaked outside of the scope of this if clause. Please consider doing something like this:
```
msg, err := s.GetMessage(pos+1)
if err != nil {
   ....
}
msgForPrefetCh = msg
```

Reply:
Nice catch, done!

Btw is there a way to catch these types of issue via some linter?

Reply:
I guess we could implement a linter for that, not sure it's worth the effort though. Moreover we do this quite a bit throughout the codebase and ideally we'd address those first.